src/samples/WorkflowCore.Sample19/Program.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using Microsoft.Extensions.Logging.Console;
+using WorkflowCore.Interface;
+using WorkflowCore.Sample19.Middleware;
+using WorkflowCore.Sample19.Steps;
+
+namespace WorkflowCore.Sample19
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var serviceProvider = ConfigureServices();
+
+            // Start the workflow host
+            var host = serviceProvider.GetService<IWorkflowHost>();
+            host.RegisterWorkflow<FlakyConnectionWorkflow, FlakyConnectionParams>();
+            host.Start();
+
+            var workflowParams = new FlakyConnectionParams
+            {
+                Description = "Flaky connection workflow"
+            };
+            var workflowId = host.StartWorkflow("flaky-sample", workflowParams).Result;
+            Console.WriteLine($"Kicked off workflow {workflowId}");
+
+            Console.ReadLine();
+            host.Stop();
+        }
+
+        private static IServiceProvider ConfigureServices()
+        {
+            // Setup dependency injection
+            IServiceCollection services = new ServiceCollection();
+            services.AddWorkflow();
+
+            // Add step middleware
+            // Note that middleware will get executed in the order in which they were registered
+            services.AddWorkflowStepMiddleware<AddMetadataToLogsMiddleware>();
+            services.AddWorkflowStepMiddleware<PollyRetryMiddleware>();
+
+            // Add some pre workflow middleware
+            // This middleware will run before the workflow starts
+            services.AddWorkflowMiddleware<AddDescriptionWorkflowMiddleware>();
```

Comment:
Should we have some mechanism for defining the priority / order the middleware executes?

Reply:
I was thinking about that as well, but it would introduce some additional complexity and also probably add some additional fields to the interface. In theory, I could have some kind of field like the following on the interface:

```cs
int? Order { get; }
```

However, then every consumer would be forced to implement it even in cases where they don't care about the order. Do you feel the additional complexity is warranted? Are there any use-cases where just changing the order in which the middleware are defined is not sufficient?

Reply:
Perhaps we leave it as a possibility for future versions?

Reply:
Yeah, sounds good. So leave it as is for now?